Feature/price updater (#83)
hermeznetwork/docs: docs/developers/price-updater.md

# Price Updater

> [Price Updater](https://github.com/hermeznetwork/price-updater-service/) is a web service to consult and update token and fiat currency used by Hermez Node.

Reviewer: ```suggestion
[Price Updater](https://github.com/hermeznetwork/price-updater-service/) is a web service used to consult and update the token and fiat currency used by Hermez Node.
```

## Installation
```
$ git clone git@github.com:hermeznetwork/price-updater-service.git
$ cd price-updater-service/
$ go build -o priceupdater # or other name that you want
```

## Pre-requirements

> 1. It is necessary to have write access to a Hermez node data base. Price updater will update the prices and write to the database.

Reviewer: ```suggestion
1. It is necessary to have write access to a Hermez node database.  The Price updater will update the prices and write them to the database.
```

> 2. You need a API Key from `https://exchangeratesapi.io/` 

Reviewer: ```suggestion
2. You need an API Key from `https://exchangeratesapi.io/` 
```

## Usage

> 1. Configure `.env` file in `price-updater` main folder. Below there is an example of and `.env` file.

Reviewer: ```suggestion
1. Configure `.env` file in the `price-updater` main folder. Below there is an example of and `.env` file.
```

POSTGRES_MAX_ID_CONNS=10
POSTGRES_MAX_OPEN_CONNS=10

# ETHEREUM NODE URL
ETH_NETWORK="http://localhost:8545"

# ROLLUP SMART CONTRACT ADDRESS
ETH_HEZ_ROLLUP="0xA68D85dF56E733A06443306A095646317B5Fa633"
ETH_USDT_ADDRESS="0xa1A31DE489C9b977fa78c70C7f001da181e126FB"

# API KEY from https://exchangeratesapi.io/`. 
FIAT_API_KEY="ffffffffffffffffffffffffffff"
BBOLT_LOCATION="priceupdater.db"
MAIN_TIME_TO_UPDATE_PRICES=1m
```

`POSTGRES_USER`, `POSTGRES_PASSWORD`, `POSTGRES_HOST` and `POSTGRES_PORT` can be extracted from the Hermez node configuration file

2. Configure provider priority. Providers are selected in order. 
```
$ ./priceupdater change-priority --priority "bitfinex,coingecko,uniswap"
```
Price Updater provides configurations files to use Bitfinex, Coingecko and Uniswap as providers.
3. Update provided configuration
```
$ ./priceupdater update-config --provider bitfinex --configFile assests/mainnet/bitfinex.json
$ ./priceupdater update-config --provider coingecko --configFile assests/mainnet/coingecko.json 
$ ./priceupdater update-config --provider uniswap --configFile assests/mainnet/uniswap.json 
```

4. Set up an apiKey to accept incoming requests
```

> $ ./priceupdater setup-apikey --apiKey "pr1c3upd4t3rw --apiKey "pr1c3upd4t3r"

Reviewer 2: ```suggestion
$ ./priceupdater setup-apikey --apiKey pr1c3upd4t3rw
```

Reviewer 2: @druiz0992
